### PR TITLE
Client context test peer

### PR DESCRIPTION
**Pull request by shangwu-google**
A mock up class for testing client context.

@vjpai

### CODE REVIEW

File: include/grpcpp/test/client_context_test_peer.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace grpc {
+namespace testing {
+
+/// A test-only class to access private members and methods of ClientContext.
+class ClientContextTestPeer {
+ public:
+  explicit ClientContextTestPeer(ClientContext* const ctx) : ctx_(ctx) {}
+
+  /// Inject metadata to the ClientContext for the test. The test spouse
```
Comment (vjpai) [inline include/grpcpp/test/client_context_test_peer.h:34]: peer

File: include/grpcpp/test/client_context_test_peer.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  /// ##must be alive when \ a ClientContext::recv_initial_metadata is called.
```
[the remark this answers is not in the document]
Comment (vjpai) [reply include/grpcpp/test/client_context_test_peer.h:35]: Also the method is called GetServerInitialMetadata

**Review by vjpai — CHANGES_REQUESTED** [review]
In addition to my comments, do you have a unit test for the test peer? I think we have a test peer unit test for the ServerContextTestSpouse.

File: include/grpcpp/test/client_context_test_peer.h (hunk 1 above)
Comment (shangwu-google) [reply include/grpcpp/test/client_context_test_peer.h:34]: Done. Thanks for the catch.

File: include/grpcpp/test/client_context_test_peer.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (shangwu-google) [reply include/grpcpp/test/client_context_test_peer.h]: Done. Thanks.

**Comment by shangwu-google** [comment]
@vjpai Unit test for client_context_test_peer.h is added. Thanks for the attention.

**Review by vjpai — APPROVED** [review]
LGTM. grpc::testing is not considered API, so this is ok for merging without a gRFC.

**Comment by vjpai** [comment]
Have you run `tools/buildgen/generate_projects.sh` ? Looks like some parts of it are still missing.

**Comment by vjpai** [comment]
Following up: what is the status here? It looks like the change requested in my previous commit was never added, and the PR is now in a conflict state.

**Comment by shangwu-google** [comment]
> Have you run `tools/buildgen/generate_projects.sh` ? Looks like some parts of it are still missing.

Could you take another look? I've fixed the conflict part and the system check is passed. However, when I ran `tools/buildgen/generate_projects.sh`, I got quite a few errors related to grpc_python_dependencies as follows
```
error loading package 'src/proto/grpc/testing': Unable to find package for @grpc_python_dependencies//:requirements.bzl: The repository '@grpc_python_dependencies' could not be resolved. and referenced by '//test/cpp/util:metrics_server_lib'
```

**Comment by lidizheng** [comment]
@shangwu-google Many build failed due to compilation errors, PTAL.

**Comment by vjpai** [comment]
This is building now. I think it's mergeable.

**Comment by vjpai** [comment]
@nupurkot I'm going to merge this.

**Comment by nupurkot** [comment]
> @nupurkot I'm going to merge this.

Thanks @vjpai for taking care of this!

**Pull request merged**